sys/src/9/port/devpci.c

```diff
@@ -58,10 +64,32 @@ pcidirgen(Chan *c, int t, int tbdf, Dir *dp)
 			BUSBNO(tbdf), BUSDNO(tbdf), BUSFNO(tbdf));
 		devdir(c, q, up->genbuf, 128, eve, 0664, dp);
 		return 1;
+	case Qpcicap:
+		p = pcimatchtbdf(tbdf);
+		if((p == nil) || (p->capcnt == 0))
+			return 0;
+		snprint(up->genbuf, sizeof up->genbuf, "%d.%d.%dcap",
+			BUSBNO(tbdf), BUSDNO(tbdf), BUSFNO(tbdf));
+		q.type = QTDIR;
+		devdir(c, q, up->genbuf, 0, eve, DMDIR|0555, dp);
+		return 1;
 	}
 	return -1;
 }
 
+/*
+ * DMG 06/13/2016. Generate the contents of the capabilities directory
+ * using the capabilities index built at the time of the PCI bus scan.
+ * Each capability is displayed as a file of length equal to the length
+ * of the capability in the memory, and the file name is formatted as
+ * capN.vV.lL.tT.bB.oO where N is capability index as collected during
+ * the PCI bus scan, V is capability vendor code (cap_vndr), L is
+ * capability length in the PCI config space (cap_len), T is capability
+ * config type (cfg_type), B is BAR number, and O is offset within BAR.
+ * The capabilities files cannot be read or written. They are displayed
+ * for exploratory purposes only.
+ */
```

Comment:
The problem is this comment only explains one part of what the function does. Can you change it to indicate all the things it does? It would be helpful for others. OR, move this comment down to line 125 to indicate how caps are presented.